### DISCUSSION TITLE
Additions to core graph styling!

### DISCUSSION THREAD

**Discussion by Acumane** [foambubble/foam#958] [Ideas]
I'd like this ticket to be a future-facing discussion following the likes of #436, from which many of the `foam.graph.style` settings were born. I'll update the following lists to reflect the conversation below.

Note: @riccardoferretti & team intends for foam-core to be an extensible, minimal foundation upon which more niche, ancillary features can be built. Suggestions should therefore be considered for their compatibility with the vanilla experience. 

To my knowledge, this is the current state of `foam.graph.style`s exposed to the user:
```json
"foam.graph.style": {
    "background": "#ffffff",
    "fontSize": 12,
    "lineColor": "#ffffff",
    "lineWidth": 0.2,
    "particleWidth": 1.0,
    "highlightedForeground": "#ffffff",
    "node": {
        "note": "#ffffff",
        "placeholder": "#ffffff",
        "type_name": "#ffffff",
        "tag_name": "#ffffff",
    }
}
```
## Suggestions:
under `"node"`, I'd love color settings for:
- orphans (no links to or from node)
- Files within a certain folder (e.g. `"notes":`, `"notes/etc":`)
- > attachments (currently we don't show attachments) —#436 
- files included within folders under `"foam.files.ignore"`; for those that wish to maintain seperation between zettels and conventional, verbose notes... the latter would be hidden on the graph until referenced by a zettel node.

### ⚠️ May violate the vanilla foam-core ideology:
* A scale modifier value dependent on:
    * number of connections (forward + backlinks)
    * tag/type

**Comment by riccardoferretti** [comment]
Thanks @Acumane, here are my comments:

I could see a setting there for "orphan" (just as we have one for "placeholder").

> Files within a certain folder (e.g. "notes":, "notes/etc":)

That to me should happen by applying a type to a node using the Foam API (once released).
At that point we could just use the existing pattern to style them.

> attachments

I could see a setting there for "attachment" (just as we have one for "placeholder"/"orphan").

> files included within folders under "foam.files.ignore";

To Foam those files don't exist, I don't think they should be included in the graph (or Foam) in any shape or form.
To me the solution to your use case is using note types in the first place (either directly in the note, or by the same API based solution as above)

> A scale modifier

Interesting idea

**Comment by chris-patenaude** [comment]
I'd like to tag onto this the ability to change the increment of growth for nodes with many edges. Currently, nodes will grow in size as edges are added, which is great. I would like to set the increment of growth so I can make it more obvious which nodes I really need to pay attention to. 

```json
// for example, if size-increment is 2, then whenever a node gets another edge it will grow by 2 units. 
"foam.graph.style": {
  "node": {
    "types": {
      "note": "#FFF",
      "placeholder": "#FFF",
      "ect.": "..."
    },
    "size-increment": 2
  }
}
```

**Reply by Acumane** [reply]
That's what I'm hoping for. Currently, the scale difference is really only noticeable on nodes w/ 10+ links. Anyone maintaining a proper Zettelkasten won't have central/parent nodes with tons of backlinks.

If added, though, the growth rate also needs a falloff (no need for a setting). Just can't have the biggest parents filling up the screen.

**Comment by kosirm** [comment]
Please add color by folder.

**Comment by kosirm** [comment]
Please add search to graph
